### ISSUE TITLE
Ktor 1.5.0 causes Kweb to show a blank page with no evident errors, Ktor 1.4.3 works fine

### ISSUE THREAD

**Issue by MrPowerGamerBR** [kwebio/kweb-core#170]
**Describe the bug**
I've cloned the [template repo](https://github.com/kwebio/kweb-template), changed Kweb's version to `0.8.2` and ran the application... and then nothing happens! The web server is up but accessing the page does nothing.

**To Reproduce**
1. Clone the cloned the [template repo](https://github.com/kwebio/kweb-template)
2. Open `build.gradle`
3. Change `kweb-core` version to 0.8.2
4. Try going to the website
5. The page is empty + Firefox shows "Issues while loading the page" on the page title + Viewing the page source code shows nothing + Trying to `curl` the page also does nothing.

Using `0.7.12`, `0.8.0` and `0.8.1` works fine, only `0.8.2` does not. No errors in the console, using Windows 10. Changing Kotlin's version to `1.4.21` also does not fix the issue.

```
2020-12-24 15:27:37.643 [main] INFO  kweb.WebBrowser - Initializing Kweb listening on port 16097
2020-12-24 15:27:37.645 [main] WARN  kweb.WebBrowser - Debug mode enabled, if in production use KWeb(debug = false)
2020-12-24 15:27:37.716 [main] INFO  org.eclipse.jetty.util.log - Logging initialized @371ms to org.eclipse.jetty.util.log.Slf4jLog
2020-12-24 15:27:37.826 [main] INFO  Application - No ktor.deployment.watch patterns match classpath entries, automatic reload is not active
2020-12-24 15:27:37.935 [main] INFO  Application - Responding at http://0.0.0.0:16097
2020-12-24 15:27:37.937 [main] INFO  org.eclipse.jetty.server.Server - jetty-9.4.31.v20200723; built: 2020-07-23T17:57:36.812Z; git: 450ba27947e13e66baa8cd1ce7e85a4461cacc1d; jvm 15.0.1+9
2020-12-24 15:27:38.013 [main] INFO  o.e.jetty.server.AbstractConnector - Started ServerConnector@38102d01{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:16097}
2020-12-24 15:27:38.013 [main] INFO  org.eclipse.jetty.server.Server - Started @670ms
2020-12-24 15:27:38.014 [main] INFO  kweb.WebBrowser - KWeb is listening on port 16097
```

**Expected behavior**
Well... I guess that the expected behavior is that it should work fine 😋

(Also, now that I noticed that I do have a pending invitation to the Kweb organization... I guess you gave me that invite when I tried using Kweb a long time (2+ years!) ago, didn't go with Kweb at the time but I tried giving it a spin now)

**Comment by MrPowerGamerBR** [comment]
Something that I've noticed:

This doesn't work with `0.8.2`: Shows an empty page
```kotlin
fun main(args: Array<String>) {
    Kweb(port = 16097, buildPage = {
        doc.body.new {
            p().text("What is your name?")
            val clickMe = input(type = InputType.text)
            val nameKVar = KVar("Peter Pan")
            clickMe.value = nameKVar
            p().text(nameKVar.map { n -> "Hi $n!" })
        }
    })
}
```

However registering the Kweb feature in Ktor works fine in `0.8.2` and I can access and use the page without any issues!
```kotlin
fun main(args: Array<String>) {
    embeddedServer(Jetty, port = 16097) {
        install(WebSockets) {}
        install(Kweb) {}

        routing {
            get("/") {
                call.respondText("hello, static!")
            }

            get("/kweb") {
                call.respondKweb {
                    this.doc.body {
                        p().text("What is your name?")
                        val clickMe = input(type = InputType.text)
                        val nameKVar = KVar("Peter Pan")
                        clickMe.value = nameKVar
                        p().text(nameKVar.map { n -> "Hi $n!" })
                    }
                }
            }
        }
    }.start(true)
}
```

**Comment by sanity** [comment]
Hmm, that's not good, will investigate tomorrow.

**Comment by sanity** [comment]
Looks like it's caused by upgrading to the latest Ktor version, but I'm not sure why it wasn't caught by the integration tests. 

While I investigate I'm releasing 0.8.3 with a downgraded Ktor.

**Comment by kokorins** [comment]
Seems same issue with 8.8 and ktor 1.5.2

**Comment by sanity** [comment]
Hmm, can't seem to replicate with Ktor 1.5.2 on Windows 10.  @kokorins What platform are you using?

**Comment by sanity** [comment]
Note: Ktor 1.5.3 recently released https://ktor.io/changelog/1.5/

**Comment by kokorins** [comment]
So, probably it's not exactly the same issue, as I'm trying to locate the issue. it looks like the `button("").on.click` is not working with ktor 1.5.* (just checked with 1.5.3).

**Comment by kokorins** [comment]
`on.click` works in kwebio 0.8.9, but now I have internal kweb error:
```
ERROR kweb.Kweb - Exception while receiving websocket message
java.lang.IllegalStateException: fromJson(json, typeToken<T>()) must not be null
        at kweb.html.events.OnReceiver$click$$inlined$event$1.invoke(OnReceiver.kt:164)
        at kweb.html.events.OnReceiver$click$$inlined$event$1.invoke(OnReceiver.kt:14)
        at kweb.html.events.OnReceiver$event$1.invoke(OnReceiver.kt:18)
        at kweb.html.events.OnReceiver$event$1.invoke(OnReceiver.kt:14)
        at kweb.Element$addEventListener$1.invoke(Element.kt:368)
        at kweb.Element$addEventListener$1.invoke(Element.kt:20)
        at kweb.Kweb.listenForWebsocketConnection(Kweb.kt:306)
        at kweb.Kweb$installRequiredKwebComponents$2$1.invokeSuspend(Kweb.kt:259)
        at kweb.Kweb$installRequiredKwebComponents$2$1.invoke(Kweb.kt)
        at io.ktor.websocket.RoutingKt.handleServerSession(Routing.kt:248)
        at io.ktor.websocket.RoutingKt.proceedWebSocket(Routing.kt:233)
        at io.ktor.websocket.RoutingKt$webSocket$2.invokeSuspend(Routing.kt:197)
        at io.ktor.websocket.RoutingKt$webSocket$2.invoke(Routing.kt)
        at io.ktor.websocket.RoutingKt$webSocketRaw$2$1$1$1$1.invokeSuspend(Routing.kt:104)
        at io.ktor.websocket.RoutingKt$webSocketRaw$2$1$1$1$1.invoke(Routing.kt)
        at io.ktor.http.cio.websocket.RawWebSocketKt.start(RawWebSocket.kt:90)
        at io.ktor.websocket.WebSocketUpgrade$upgrade$2.invokeSuspend(WebSocketUpgrade.kt:98)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
        at java.base/java.lang.Thread.run(Thread.java:834)

```

**Comment by sanity** [comment]
Yes, we're aware of it - #189, it doesn't look like it affects functionality. @Derek52 is working on #59 which we believe will fix this or at least help to identify the problem.

**Comment by sanity** [comment]
Should be fixed in `0.9.0`.
